### DISCUSSION TITLE
Cover: attribute year / year of replacement / year of rehabilitation

### DISCUSSION THREAD

**Discussion by kandre** [QGEP/QGEP#746] [Q&A] [answered]
Our netwok manager would need to record the year, year of replacement or year of rehabilitation for manholes covers for an easier maintenance.

As far as I know from the VSA and QGEP model, there is no such attributes on covers.
Does anyone knows where we could / shoud store those information ?

**Comment by sjib** [comment, marked as answer]
You could use the maintenance_event for this and create the following events:
n-times:
cover_replaced_year_19xx (and connect all wastewater_structures, where the cover was replaced in the year 19xx), type = choose one of the values, status = done

m-times:
cover_rehabilitation_planned_year_20xx (and connect all wastewater_structures, where the cover is planned to be replaced in the year 20_xx), type =  choose one of the values, status = planned

And maybe you will have around the current date both because you planned to replace the covers but did not do it (yet) - so you will see in cover_rehabilitation_planned_year_2019 what you did not do yet this year and in cover_replaced_year_2019 what you really did.

This means also that you will have to move fixed covers from one event to the other category as a workflow.
